services/update-faq/update-faq.service.js

```diff
@@ -0,0 +1,56 @@
+const { EmbedBuilder } = require('discord.js');
+const config = require('../../config');
+
+class UpdateFAQsService {
+  static EmbedDescriptionCharacterLimit = 4096;
+
+  static async handleInteraction(interaction) {
+    let rawFAQs;
+    try {
+      rawFAQs = await UpdateFAQsService.fetchFAQs();
+    } catch (error) {
+      console.log(error);
+      await interaction.reply('Failed to update FAQs');
+      return;
+    }
+
+    const FAQEmbeds = UpdateFAQsService.createFAQEmbeds(rawFAQs);
+    const FAQChannel = interaction.guild.channels.cache.get(config.channels.FAQChannelId);
+    await UpdateFAQsService.deletePreviousFAQs(FAQChannel);
+    await UpdateFAQsService.sendFAQs(FAQEmbeds, FAQChannel);
+    await interaction.reply('FAQs updated');
+  }
+
+  static async fetchFAQs() {
+    const response = await fetch('https://raw.githubusercontent.com/TheOdinProject/top-meta/main/FAQ.md');
+    const FAQs = await response.text();
+    return FAQs;
+  }
+
+  static createFAQEmbeds(rawFAQs) {
+    return UpdateFAQsService
+      .segments(rawFAQs, UpdateFAQsService.EmbedDescriptionCharacterLimit)
+      .map((chunk) => new EmbedBuilder().setColor('#cc9543').setDescription(chunk));
+  }
+
+  static async sendFAQs(FAQEmbeds, FAQChannel) {
+    await Promise.all(FAQEmbeds.map(async (e) => {
+      await FAQChannel.send({ embeds: [e] });
+    }));
+  }
+
+  static async deletePreviousFAQs(FAQChannel) {
+    const prev = await FAQChannel.messages.fetch({ limit: 10 });
+    prev.forEach((message) => message.delete());
```

Comment:
Should this contain a check for whether or not the channel has messages (ie if `FAQChannel.messages` isn't `undefined`). Right now this will give errors if the channel contains no messages at all, which can only be solved by sending a new message, and then recreating the cache.

Given that, while I don't expect there ever to be no messages, might still be worth to add a check for them

Reply:
Thanks for commenting @Asartea 

---


>  if `FAQChannel.messages` isn't `undefined`

I don't think the `messages` property will ever be undefined. It returns a [`MessageManager` according to the docs](https://old.discordjs.dev/#/docs/discord.js/14.9.0/class/TextChannel?scrollTo=messages).

> Right now this will give errors if the channel contains no messages at all

I haven't encountered any errors so far; I'd love if you can also look into this. If there are no messages, the `fetch` method just returns a `Collection` with 0 size. The `forEach` iterator gets skipped altogether in this case.

> then recreating the cache

I'm not sure what is meant by cache here?

Reply:
Huh, fascinatingly enough I can't reproduce the error anymore; managed to get it earlier when I was testing it, but now it seems fine. Might be that it somehow hadn't picked the channel I was using for testing yet. Can't reproduce the error anymore, so considering this resolved